PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -188,8 +198,92 @@ PATMuonProducer::~PATMuonProducer()
 {
 }
 
+GlobalPoint* PATMuonProducer::getMuonDirection(const reco::MuonChamberMatch& chamberMatch,
+						const edm::ESHandle<GlobalTrackingGeometry>& geometry,
+						const DetId& chamberId)
+{
+  const GeomDet* chamberGeometry = geometry->idToDet( chamberId );
+  if (chamberGeometry){
+    LocalPoint localPosition(chamberMatch.x, chamberMatch.y, 0);
+    return new GlobalPoint(chamberGeometry->toGlobal(localPosition));
+  }
+  return nullptr;
+
+}
+
+
+void PATMuonProducer::fillL1TriggerInfo(pat::Muon& aMuon,
+					edm::Handle<std::vector<pat::TriggerObjectStandAlone> >& triggerObjects,
+					const edm::ESHandle<GlobalTrackingGeometry>& geometry)
+{
+  // L1 trigger object parameters are defined at MB2/ME2. Use the muon
+  // chamber matching information to get the local direction of the
+  // muon trajectory and convert it to a global direction to match the
+  // trigger objects
+
+  GlobalPoint* muonPosition(nullptr);
+  // loop over chambers
+  for ( const auto& chamberMatch: aMuon.matches() ) {
+    if ( chamberMatch.id.subdetId() == MuonSubdetId::DT) {
+      DTChamberId detId(chamberMatch.id.rawId());
+      if (abs(detId.station())!=2) continue;
+      muonPosition = getMuonDirection(chamberMatch, geometry, detId);
+      break;
+    }
+    if ( chamberMatch.id.subdetId() == MuonSubdetId::CSC) {
+      CSCDetId detId(chamberMatch.id.rawId());
+      if (abs(detId.station())!=2) continue;
+      muonPosition = getMuonDirection(chamberMatch, geometry, detId);
+      break;
+    }
+  }
+  if (not muonPosition) return;
+  for (unsigned int i=0; i<triggerObjects->size(); ++i){
+    if (triggerObjects->at(i).hasTriggerObjectType(trigger::TriggerL1Mu)){
+      if (deltaR(triggerObjects->at(i).p4(),*muonPosition)>0.1) continue;
+      // printf("muon pt: %4.1f eta: %+5.3f phi: %+5.3f\n", aMuon.pt(), aMuon.eta(), aMuon.phi());
+      // if (muonPosition) printf("global direction MB2/ME2: eta: %+5.3f phi: %+5.3f\n",double(muonPosition->eta()), double(muonPosition->phi()));
+      // printf("L1 pt: %4.1f eta: %+5.3f phi: %+5.3f\n",triggerObjects->at(i).pt(), triggerObjects->at(i).eta(), triggerObjects->at(i).phi());
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant